TESTS/mbed_hal/ticker/main.cpp

```diff
@@ -1986,6 +1986,34 @@ static void test_irq_handler_insert_non_immediate_in_irq()
     TEST_ASSERT_EQUAL(0, interface_stub.disable_interrupt_call);
 }
 
+static uint32_t ticker_interface_stub_read_interrupt_time()
+{
+    ++interface_stub.read_call;
+    // only if set interrupt call, to test the condition afterwards
+    if (interface_stub.set_interrupt_call) {
+        return interface_stub.interrupt_timestamp;
+    } else {
+        interface_stub.timestamp;
+    }
+} 
+
+static void test_set_interrupt_past_time()
+{
+    interface_stub.read_call = 0;
+    interface_stub.set_interrupt_call = 0;
+    interface_stub.timestamp = 0xFF;
+
+    interface_stub.interface.read = ticker_interface_stub_read_interrupt_time;
+
+    ticker_event_t event = { 0 };
+    const timestamp_t event_timestamp = interface_stub.timestamp;
```

Comment:
Could you add another test similar to this one where `event_timestamp` is  more than `interface_stub.timestamp`.